src/Common/src/TypeSystem/Ecma/EcmaModule.cs

```diff
@@ -480,47 +477,15 @@ public override TypeDesc GetGlobalModuleType()
             return GetType(MetadataTokens.EntityHandle(0x02000001 /* COR_GLOBAL_PARENT_TOKEN */));
         }
 
-        private static AssemblyContentType GetContentTypeFromAssemblyFlags(AssemblyFlags flags)
+        protected static AssemblyContentType GetContentTypeFromAssemblyFlags(AssemblyFlags flags)
         {
             return (AssemblyContentType)(((int)flags & 0x0E00) >> 9);
         }
 
-        private AssemblyName _assemblyName;
-
-        // Returns cached copy of the name. Caller has to create a clone before mutating the name.
-        public AssemblyName GetName()
-        {
-            if (_assemblyName == null)
-            {
-                AssemblyName an = new AssemblyName();
-                an.Name = _metadataReader.GetString(_assemblyDefinition.Name);
-                an.Version = _assemblyDefinition.Version;
-                an.SetPublicKey(_metadataReader.GetBlobBytes(_assemblyDefinition.PublicKey));
-
-                an.CultureName = _metadataReader.GetString(_assemblyDefinition.Culture);
-                an.ContentType = GetContentTypeFromAssemblyFlags(_assemblyDefinition.Flags);
-
-                _assemblyName = an;
-            }
-
-            return _assemblyName;
-        }
-
         public string GetUserString(UserStringHandle userStringHandle)
         {
             // String literals are not cached
             return _metadataReader.GetUserString(userStringHandle);
         }
-
-        public bool HasCustomAttribute(string attributeNamespace, string attributeName)
-        {
-            return _metadataReader.GetCustomAttributeHandle(_assemblyDefinition.GetCustomAttributes(),
-                attributeNamespace, attributeName).IsNil;
-        }
-
-        public override string ToString()
```

Comment:
I think there needs to be an implementation of ToString for EcmaModule. Especially as the ToString on EcmaType will use it